### ISSUE TITLE
KeyError: 'donor_method'

### ISSUE THREAD

**Issue by sarahbonnin** [czbiohub/PopV#22]
Hello,

Thank you for developing this great tool.

I am testing this template:
https://colab.research.google.com/github/czbiohub/PopV/blob/main/tabula_sapiens_tutorial.ipynb

with a skin dataset.

At the "Process_Query" step, I get an error:

`KeyError: 'donor_method'`

This corresponds to the `ref_batch_key` which we give as `donor_assay` in the template. The ref_labels_key parameter has `donor_method` set as default:

```
class Process_Query:
    def __init__(
        self,
        query_adata: anndata.AnnData,
        ref_adata: anndata.AnnData,
        ref_labels_key: str = "cell_ontology_class",
        ref_batch_key: str = "donor_method",
...
```

But it does not work when I assign `donor_assay` explicitely:

```
from popv.preprocessing import Process_Query

adata = Process_Query(
    query_adata,
    ref_adata,
    query_labels_key=query_labels_key,
    query_batch_key=query_batch_key,
    ref_labels_key=ref_labels_key,
    ref_batch_key="donor_assay",
    unknown_celltype_label=unknown_celltype_label,
    save_path_trained_models=output_model_fn,
    #cl_obo_folder="./PopV/ontology/",
    cl_obo_folder="./ontology",
    #prediction_mode="inference",  # 'fast' mode gives fast results (does not include BBKNN and Scanorama and makes more inaccurate errors)
    n_samples_per_label=n_samples_per_label,
    use_gpu=0,
    compute_embedding=True,
    hvg=None
).adata
```

I am not sure how to fix this, although it might be very obvious.

Unrelated: I had to change a line of code (it is perhaps only a git version issue, but just in case):
the following did not work:
`!git clone https://github.com/czbiohub/PopV.git@fast_popv`
so I changed to:
`!git clone -b fast_popv https://github.com/czbiohub/PopV.git`

Thank you!

Best,
Sarah

**Comment by cane11** [comment]
Hi Sarah, could you install the main branch? This should be fixed.

**Comment by sarahbonnin** [comment]
Thank you!

I did clone the master branch instead. I made sure I clone only the master branch ("main" branch) with:
`!git clone -b main --single-branch https://github.com/czbiohub/PopV.git`

But I now realized it is an error on my end: I was focused on the `ref_batch_key` parameter, while in a previous cell in the template script, `query_batch_key` is set to "donor_method", which I do not have in my data set.

The preprocessing now complete.

My apologies, and thank you for your time!
